crates/core/src/buc/connection.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{
+	cnf::{GLOBAL_BUCKET, GLOBAL_BUCKET_ENFORCED},
+	err::Error,
+};
+use dashmap::DashMap;
+use std::sync::Arc;
+
+use super::store::{prefixed::PrefixedStore, Key, ObjectStore};
+
+// Helper type to represent how bucket connections are persisted
+pub type BucketConnections = DashMap<(String, String, String), Arc<dyn ObjectStore>>;
+
+/// Connect to a global bucket, if one is configured
+/// If no global bucket is configured, the NoGlobalBucket error will be returned
+/// The key in the global bucket will be: `{ns}/{db}/{bu}`
+pub fn connect_global(ns: &str, db: &str, bu: &str) -> Result<Arc<dyn ObjectStore>, Error> {
+	// Obtain the URL for the global bucket
+	let Some(ref url) = *GLOBAL_BUCKET else {
+		return Err(Error::NoGlobalBucket);
+	};
+
+	// Connect to the global bucket
+	let global = connect(url, true, false)?;
+
+	// Create a prefixstore for the specified bucket
+	let key = Key::from(format!("/{ns}/{db}/{bu}"));
+	Ok(Arc::new(PrefixedStore::new(global, key)))
+}
+
+/// Connects to a bucket by it's connection URL
+/// The function:
+/// - Checks if the global bucket is enfored
```

Comment:
```suggestion
/// - Checks if the global bucket is enforced
```

Reply:
Done in [120e55e](https://github.com/surrealdb/surrealdb/pull/5701/commits/120e55e0ad47bb91671a7d96931d22c5488e2b96)